### PR TITLE
[BYOC] Support Tuple Output in C/DNNL Codegen

### PR DESCRIPTION
**Pull request by comaniac**
We previously supported multiple outputs in graph partition pass by creating a new tuple node as a single output. However, we missed a point that this tuple output has to be decomposed when generating C code.

Fortunately, TVM runtime already flats tuples to multiple tensors, so it's not hard to make runtime working. This PR modifies the C codegen accordingly. Note that although the new added unit test uses C codegen, I've tested locally and make sure that this change is also applicable to DNNL.

cc @masahi (sorry for the confusions 😓 ), @zhiics

### CODE REVIEW

File: tests/python/relay/test_pass_partition_graph.py [hunk 1]
```diff
@@ -201,8 +202,11 @@ def check_vm_result():
         exe = runtime.vm.Executable.load_exec(code, lib)
         vm = runtime.vm.VirtualMachine(exe)
         vm.init(ctx)
-        out = vm.run(**map_inputs)
-        tvm.testing.assert_allclose(out.asnumpy(), result, rtol=tol, atol=tol)
+        outs = vm.run(**map_inputs)
+        outs = outs if len(outs) > 1 else [outs]
```
Comment (zhiics) [inline tests/python/relay/test_pass_partition_graph.py:206]: is `len(outs)` oaky? It doesn't necessarily be an ADTObj/array, right? what if it is an objectref (i.e. NDArray in this case)?

**Comment by zhiics** [comment]
@comaniac Thanks for adding this support. I always thought this was done yet...

File: tests/python/relay/test_pass_partition_graph.py (hunk 1 above)
Comment (comaniac) [reply tests/python/relay/test_pass_partition_graph.py:206]: Yeah the test failed because of this. Changing to `isinstance(outs, runtime.container.ADT)`

**Comment by masahi** [comment]
Thanks @comaniac @zhiics

**Pull request merged**